pkg/volume/empty_dir/empty_dir_quota.go

```diff
@@ -0,0 +1,104 @@
+package empty_dir
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+var _ volume.VolumePlugin = &EmptyDirQuotaPlugin{}
+var _ volume.Builder = &emptyDirQuotaBuilder{}
+
+// EmptyDirQuotaPlugin is a simple wrapper for the k8s empty dir plugin builder.
+type EmptyDirQuotaPlugin struct {
+	// wrapped is the actual k8s emptyDir volume plugin we will pass method calls to.
+	Wrapped volume.VolumePlugin
+
+	// The default quota to apply to each node:
+	DefaultQuota resource.Quantity
```

Comment:
No, it was planned but scrapped during development, I will rename that to just Quota.